scripts/dts/gen_defines.py

```diff
@@ -164,10 +164,14 @@ def write_device_extern_header(device_header_out, edt):
         print("#endif", file=dev_header_file)
         print("", file=dev_header_file)
         print("#include <zephyr/device.h>", file=dev_header_file)
+        print("#include <zephyr/drivers/emul.h>", file=dev_header_file)
 
         for node in sorted(edt.nodes, key=lambda node: node.dep_ordinal):
             print(f"extern const struct device DEVICE_DT_NAME_GET(DT_{node.z_path_id}); /* dts_ord_{node.dep_ordinal} */",
                   file=dev_header_file)
+            print(f"extern const struct emul EMUL_DT_NAME_GET(DT_{node.z_path_id});" \
+                    f" /* dts_ord_{node.dep_ordinal} */",
+                    file=dev_header_file)
```

Comment:
Like I may have mentioned, I haven't been following the emulator device development carefully, but this looks problematic to me from a layering perspective. device_extern.h is for devices, so this mixing feels off.

Is this a good place for me to start reading? https://docs.zephyrproject.org/latest/hardware/emulator/index.html

Reply:
I have updated the commit to put the `struct emul`s in their own header file and add the associated boilerplate to `gen_defines.py` and `dts.cmake`.

I thought about trying to combine `write_device_extern_header` and `write_emul_extern_header` into one parameterized function, but constructing the include guard, the name of the public header file, and the actual declaration in a parameterized way seemed error-prone and less readable.

Reply:
@alevkoy can this be marked resolved now?